#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: use /bin/sh as the ldd shell always
# DP: Related bugs: N/A
# DP: Patch author: Juha Kallioinen <juha.kallioinen@nokia.com>
# DP: Upstream status: Not submitted
# DP: Status Details: N/A
# DP: Date: 2009-06-12

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p2 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p2 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

@DPATCH@
diff -Naur old/glibc-2.5/elf/Makefile new/glibc-2.5/elf/Makefile
--- old/glibc-2.5/elf/Makefile	2007-09-25 17:20:30.000000000 +0300
+++ new/glibc-2.5/elf/Makefile	2009-06-12 12:08:08.000000000 +0300
@@ -359,15 +359,16 @@
 
 common-ldd-rewrite = -e 's%@RTLD@%$(slibdir)/$(rtld-installed-name)%g' \
 		     -e 's%@VERSION@%$(version)%g'
-sh-ldd-rewrite = $(common-ldd-rewrite) -e 's%@BASH@%/bin/sh%g;s/\$$"/"/g'
+sh-ldd-rewrite = $(common-ldd-rewrite) -e 's%@BASH@%/bin/sh%g;s/\$$"/"/g' \
+		   -e 's%@TEXTDOMAINDIR@%$(msgcatdir)%g'
 bash-ldd-rewrite = $(common-ldd-rewrite) -e 's%@BASH@%$(BASH)%g' \
 		   -e 's%@TEXTDOMAINDIR@%$(msgcatdir)%g'
 
-ifneq ($(have-bash2),yes)
+#ifneq ($(have-bash2),yes)
 ldd-shell = sh
-else
-ldd-shell = bash
-endif
+#else
+#ldd-shell = bash
+#endif
 
 ifeq ($(ldd-rewrite-script),no)
 define gen-ldd
